src/controllers/patient.ts

```diff
@@ -0,0 +1,114 @@
+import { Request, Response } from "express";
+import {
+  getAll,
+  sendAndLog,
+  throwException,
+  validateObjectId,
+  validatePatient,
+} from "helper";
+import asyncHandler from "middleware/asyncHandler";
+import Appointment from "models/appointment";
+import Patient from "models/patient";
+
+export const getOneField = asyncHandler(async (req: Request, res: Response) => {
+  const { _id, field } = req.params;
+  if (!(await validateObjectId(_id))) throwException(res, "Invalid Id", 400);
+  const result: any = await Patient.findById(_id);
+  if (!result) throwException(res, "Invalid Id", 400);
+  if (!result[field]) throwException(res, "Invalid field", 400);
```

Comment:
move if condition from this code and create a function to check null or empty array.